### DISCUSSION TITLE
Bring Back the Re-obfuscated Version of Paper

### DISCUSSION THREAD

**Discussion by Foorcee** [PaperMC/Paper#10950] [Ideas]
### Is your feature request related to a problem?

I would like to propose the reintroduction of the obfuscated version of Paper. Despite the remapping feature, we are unable to use it because not all reflection calls are supported. Users should be able to choose whether they want to use the obfuscated or unobfuscated version at runtime.

We have been very successful with the obfuscated version so far and currently do not see any added value in using the Mojang-mapped version. Remapping the plugins consumes unnecessary resources, and we do not wish to operate our plugins in the Mojang mappings.

### Describe the solution you'd like.

I understand that the Mojang-mapped version is offered and see its benefits for some users. However, it would be very helpful if at least the old obfuscated version could still be provided through the download API until the existing issues are resolved.

### Describe alternatives you've considered.

-

**Comment by Machine-Maker** [comment]
> not all reflection calls are supported

What reflection calls are not supported? If some are missing, they can be added. 

> Users should be able to choose whether they want to use the obfuscated or unobfuscated version at runtime.

This kind of choice should have 0 impact on users, it really has nothing to do with the operation of the server.

> Remapping the plugins consumes unnecessary resources

What kind of resources? Plugins are remapped once and then cached.

**Reply by Foorcee** [reply]
> What reflection calls are not supported? If some are missing, they can be added.

We have added dependencies via the `java -cp ..` in the classpath. The reflection calls of this are not remapped.

> This kind of choice should have 0 impact on users, it really has nothing to do with the operation of the server.

The term user refers to the operator.

> What kind of resources? Plugins are remapped once and then cached.

Even if the plugins are cached, additional resources are still required for remapping the reflection at runtime. This may not make much difference on a single Minecraft server, but with 100 servers the effort should not be underestimated. In addition, we currently have to ensure that the caches are preserved. So far, our servers have been thoroughly cleaned of artifacts after a restart. It is therefore generally advantageous to create as few manageable artifacts as possible.

**Comment by jpenilla** [comment]
We will not support reobfuscated Paper runtimes on 1.20.5 or newer.

**Comment by electronicboy** [comment]
You can still produce a reobf jar to my understanding (However, these are 100% unsupported), but the reality here is that we are moving away from spigot, in which, we will no longer have any obfuscation mappings to go off. For some time we might release jars containing a copy of spigots mappings after spigot releases, but it will not be a long term thing.

if there is any missing holes in the tooling, it would be best that issues are created so that we can resolve them, because we have no interest in moving backwards in our progress to separate from spigot.

**Reply by Foorcee** [reply]
I understand that there might be a desire to differentiate oneself from Spigot. However, from my perspective, this change doesn't separate oneself so much from Spigot as it does from Minecraft itself. This change moves away from how Mojang releases and operates Minecraft.

**Reply by masmc05** [reply]
Paper wants to release versions before spigot releases them, on that time they just can't provide reobfuscated versions since spigot doesn't use Minecraft obfuscated names everywhere, but own semi deobfuscated names, and they just can't guess what remappings they'll invent. That's why they move to mojang mappings, they are released with the version itself and are more reliable, so they'll be able to use them in future

**Reply by masmc05** [reply]
Current change allows devs to move completely to mojang mappings so in future they could support this, but at the same time won't need to worry if some random guy will want to still use reobfuscated paper

**Reply by jpenilla** [reply]
This discussion is over, but to provide you with some context: As of 1.21, MinecraftForge, NeoForge, Sponge, and Paper all use Mojang mappings at runtime. Spigot is an outlier in this regard.

**Comment by Foorcee** [comment]
I was already aware that we had encountered some issues with the transition and therefore wanted to suggest reconsidering it. I'm also aware that the Re-Obf Jar can still be generated. It just adds extra effort to build the project manually. Hence, the suggestion was to adjust the Download API to continue providing the Re-Obf Jar. The Plugin Remapping is already a quite specific operation, from which I haven't yet seen any added value.

**Reply by jpenilla** [reply]
Thanks for the suggestion, we have considered it and have decided not to accept it: https://github.com/PaperMC/Paper/discussions/10950#discussioncomment-9843317
It wouldn't make sense for us to provide downloads for something that we don't support.
